src/modules/components/wallet/wallet.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import { useEnsName } from 'wagmi';
+import { StateSkeletonBar } from '../../../core';
+import { type ICompositeAddress } from '../../types';
+import { addressUtils } from '../../utils';
+import { MemberAvatar } from '../member';
+
+export interface IWalletProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+    /**
+     * The connected user details.
+     */
+    user: ICompositeAddress;
+    /**
+     * Whether or not the app is connected to a wallet.
+     */
+    isConnected: boolean;
+}
+
+export const Wallet: React.FC<IWalletProps> = (props) => {
+    const { user, isConnected = false, onClick, ...otherProps } = props;
+    const { data: ensName, isLoading: ensLoading } = useEnsName({
+        address: isConnected ? addressUtils.getChecksum(user.address) : undefined,
+        query: { enabled: isConnected },
```

Comment:
The query should only be enabled if the `user.name` property is not already set and if the `user` property is set